qml/Keyboard.qml

```diff
@@ -380,20 +503,94 @@ Item {
     function sendEndKey() {
         event_handler.onKeyReleased("", "end");
     }
+    function selectLeft() {
+        event_handler.onKeyReleased("SelectPreviousChar", "keysequence");
+    }
+    function selectRight() {
+        event_handler.onKeyReleased("SelectNextChar", "keysequence");
+    }
+    function selectUp() {
+        event_handler.onKeyReleased("SelectPreviousLine", "keysequence");
+    }
+    function selectDown() {
+        event_handler.onKeyReleased("SelectNextLine", "keysequence");
+    }
+    function selectWord() {
+        event_handler.onKeyReleased("MoveToPreviousWord", "keysequence");
+        event_handler.onKeyReleased("SelectNextWord", "keysequence");
+    }
+    function selectStartOfLine() {
+        event_handler.onKeyReleased("SelectStartOfLine", "keysequence");
+    }
+    function selectEndOfLine() {
+        event_handler.onKeyReleased("SelectEndOfLine", "keysequence");
+    }
+    function selectStartOfDocument() {
+        event_handler.onKeyReleased("SelectStartOfDocument", "keysequence");
+    }
+    function selectEndOfDocument() {
+        event_handler.onKeyReleased("SelectEndOfDocument", "keysequence");
+    }
+    function selectAll() {
+        event_handler.onKeyReleased("SelectAll", "keysequence");
+    }
+    function moveToStartOfLine() {
+        event_handler.onKeyReleased("MoveToStartOfLine", "keysequence");
+    }
+    function moveToEndOfLine() {
+        event_handler.onKeyReleased("MoveToEndOfLine", "keysequence");
+    }
+    function moveToStartOfDocument() {
+        event_handler.onKeyReleased("MoveToStartOfDocument", "keysequence");
+    }
+    function moveToEndOfDocument() {
+        event_handler.onKeyReleased("MoveToEndOfDocument", "keysequence");
+    }
+    function redo() {
+        event_handler.onKeyReleased("Redo", "keysequence");
+    }
+    function undo() {
+        event_handler.onKeyReleased("Undo", "keysequence");
+    }
+    function paste() {
+        event_handler.onKeyReleased("Paste", "keysequence");
+    }
+    function copy() {
+        event_handler.onKeyReleased("Copy", "keysequence");
+    }
+    function cut() {
+        event_handler.onKeyReleased("Cut", "keysequence");
+    }
 
     function processSwipe(positionX, positionY) {
-        if (positionX < prevSwipePositionX - units.gu(1) && input_method.surroundingLeft != "") {
-            sendLeftKey();
+        // TODO: Removed input_method.surrounding* from the criteria until they are fixed in QtWebEngine
```

Comment:
Also needs a bug report attached

Reply:
https://github.com/ubports/ubuntu-touch/issues/1157

Reply:
I've added the link to the bug in the latest commit.